pincer/core/gateway.py

```diff
@@ -310,19 +311,22 @@ async def handle_reconnect(self, payload: GatewayDispatch):
             self.shard_key
         )
 
+        self.__should_resume = True
         await self.__socket.close()
-        self.__should_reconnect = True
-        await self.start_loop()
 
     async def handle_invalid_session(self, payload: GatewayDispatch):
         """|coro|
         Opcode 9 - Invalid connection
         Attempt to relog. This is probably because the session was already invalidated
         when we tried to reconnect.
         """
-        _log.debug("%s Invalid session, attempting to relog...", self.shard_key)
-        self.__should_reconnect = False
-        await self.start_loop()
+
+        if payload.data:
+            _log.debug("%s Invalid session, attempting to reconnect...", self.shard_key)
+        else:
+            _log.debug("%s Invalid session, attempting to relog...", self.shard_key)
```

Comment:
eh? not the most readable thing but feels less redundant
```suggestion
        _log.debug(
            "%s Invalid session, attempting to re%s...",
            self.shard_key, ("log", "connect")[bool(payload.data)]
        )
```

Reply:
i'd prefer a `a if b else c` ternary instead of `(c, a)[b]`

Reply:
I can switch it to a ternary. It will be the entire word though cause splitting a word in half is kinda sus.